app/config.js

```diff
@@ -1,48 +1,111 @@
+
+import htmlParser from 'htm-to-json';
 import mergeWith from 'lodash/mergeWith';
 import defaultConfig from './configurations/config.default';
 
 const configs = {}; // cache merged configs for speed
 const themeMap = {};
+const piwikMap = [];
 
 if (defaultConfig.themeMap) {
   Object.keys(defaultConfig.themeMap).forEach((theme) => {
     themeMap[theme] = new RegExp(defaultConfig.themeMap[theme], 'i'); // str to regex
   });
 }
 
+if (defaultConfig.piwikMap) {
+  for (let i = 0; i < defaultConfig.piwikMap.length; i++) {
+    piwikMap.push({
+      id: defaultConfig.piwikMap[i].id,
+      expr: new RegExp(defaultConfig.piwikMap[i].expr, 'i'),
+    });
+  }
+}
+
 function customizer(objValue, srcValue) {
   if (Array.isArray(objValue)) { return srcValue; } // Return only latest if array
   return undefined; // Otherwise use default customizer
 }
 
-export function getNamedConfiguration(configName) {
-  if (!configs[configName]) {
+export function addMetaData(config) {
```

Comment:
Good point. There is html parsing and all kinds of actions involved. Now part of config caching as you suggested